cdap-security/src/main/java/co/cask/cdap/security/zookeeper/SharedResourceCache.java

```diff
@@ -165,16 +166,15 @@ public T getIfPresent(Object key) {
   public void put(final String name, final T instance) {
     final String znode = joinZNode(parentZnode, name);
     try {
-      final byte[] encoded = codec.encode(instance);
       LOG.debug("Setting value for node {}", znode);
-      ListenableFuture<String> future = ZKExtOperations.createOrSet(zookeeper, znode, encoded, znode,
-                                                                    MAX_RETRIES, znodeACL);
+      ListenableFuture<T> future = ZKExtOperations.createOrSet(zookeeper, znode, Suppliers.ofInstance(instance),
+                                                               codec, MAX_RETRIES, znodeACL);
 
-      Futures.addCallback(future, new FutureCallback<String>() {
+      Futures.addCallback(future, new FutureCallback<T>() {
         @Override
-        public void onSuccess(String result) {
+        public void onSuccess(T result) {
           LOG.debug("Created or set node {}", znode);
-          resources.put(name, instance);
```

Comment:
No. In the old code, the createOrSet won't return the actual value be set in the node (because it won't change, you always passes in the `byte[]). However, in the new version of createOrSet, it will return the actual value being set (through the Future). Hence, it's more correct to use that value and store that in the local cache. However, in this particular usage, `result` is the same as `instance` because the `Supplier` we provided to the `createOrSet` method is a constant supplier.